eZ/Publish/Core/Repository/Helper/DomainMapper.php

```diff
@@ -136,15 +136,17 @@ public function buildDomainFields(array $spiFields, $contentType)
 
         $fields = array();
         foreach ($spiFields as $spiField) {
-            $fields[] = new Field(
-                array(
-                    'id' => $spiField->id,
-                    'value' => $this->fieldTypeRegistry->getFieldType($spiField->type)
-                        ->fromPersistenceValue($spiField->value),
-                    'languageCode' => $spiField->languageCode,
-                    'fieldDefIdentifier' => $fieldIdentifierMap[$spiField->fieldDefinitionId],
-                )
-            );
+            if (array_key_exists($spiField->fieldDefinitionId, $fieldIdentifierMap)) {
```

Comment:
rather:
```php
// We ignore fields in content not part of the content type
if (isset($fieldIdentifierMap[$spiField->fieldDefinitionId])) {
```

However even better would to iterate fields from content type, so we generate default values for missing fields, and by design ignore additional fields.

pov @pspanja ?